include/nrf_cloud.h

```diff
@@ -244,6 +244,16 @@ int nrf_cloud_sensor_attach(const struct nrf_cloud_sa_param *param);
  */
 int nrf_cloud_sensor_data_send(const struct nrf_cloud_sensor_data *param);
 
+/**
+ * @brief Update the device shadow with sensor data.
+ *
+ * @param[in] param Sensor data.
```

Comment:
Is it always sensor data? I think this function is generic and could send any data to the shadow

Reply:
Currently it is always "cloud sensor data" from the cloud module's perspective. It should be expanded in the future, but for now is this PR's scope to transmit cloud sensor data to the shadow.